src/utils/daos_hdlr.c

```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+		if (rc != 0) {
+			fprintf(stderr, "failed to parse cont uuid\n");
+			D_GOTO(out, rc);
+		}
+	} else {
+		fprintf(stderr, "failed to parse uuid or path\n");
+		D_GOTO(out, rc = EINVAL);
+	}
+out:
+	return rc;
+}
+
+static int
+fs_copy_parse(char *src,
+	      char *dst,
+	      struct fs_copy_args *fa,
+	      struct file_dfs *src_file_dfs,
+	      struct file_dfs *dst_file_dfs,
+	      struct duns_attr_t *src_dattr,
+	      struct duns_attr_t *dst_dattr,
+	      char **src_str,
+	      char **src_dfs_path,
+	      int *src_str_len,
+	      char **dst_str,
+	      char **dst_dfs_path,
+	      int *dst_str_len)
+{
+	/* parse src and dst strings
+	 * for DAOS, daos: specified in string
+	 * for POSIX, just use path
+	 * example: --src={daos://pool/cont}
+	 * example: --dst={/a/path}
+	 */
+	int	rc = 0;
+	char	*src_saveptr = NULL;
+	char	*dst_saveptr = NULL;
+	char	*src_type = NULL;
+	char	*dst_type = NULL;
+	char	*src_pool = NULL;
+	char	*src_cont = NULL;
+	char	*dst_pool = NULL;
+	char	*dst_cont = NULL;
+
+	/* validate and enforce "daos://" prefix is used, if not,
+	 * set to "path" type
+	 */
+	if (strncmp(src, "daos://", 7) == 0) {
+		src_type = "daos";
+		/* start of src pool/cont str */
+		src += 7;
+		/* daos:// should not have more than two
+		 * forward slashes (i.e. daos:///)
+		 */
+		if (strncmp(src, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos src type "
+				"format, please use: \n"
+				"\t--src=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		src_type = "path";
+	}
+	if (strncmp(dst, "daos://", 7) == 0) {
+		dst_type = "daos";
+		/* start of dst pool/cont str */
+		dst += 7;
+		if (strncmp(dst, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos dst type format, "
+				"please use: \n"
+				"\t--dst=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		dst_type = "path";
+	}
+
+	/* check for src DAOS pool/cont or UNS path */ 
+	if (strcmp(src_type, "daos") == 0) {
+		/* check if this copy is from root of DFS cont */
+		*src_str = strdup("/");
+		*src_dfs_path = strdup(src);
+		rc = fs_copy_parse_uuids(src, &src_saveptr, &(fa->src_p_uuid),
+					 &(fa->src_c_uuid), &src_pool,
+					 &src_cont);
+		if (rc != 0 || (src_pool == NULL || src_cont == NULL)) {
+			fprintf(stderr, "failed to parse source DAOS uuids\n");
+			D_GOTO(out, rc);
+		}
+		int src_len = strlen(*src_dfs_path);
+		int src_pool_cont_len = (strlen(src_pool) +
+					strlen(src_cont) + 1);
+		/* copy from root if equal */
+		if (src_len != src_pool_cont_len) {
+			/* get the container subset path */
+			*src_dfs_path += strlen(src_pool) +
+					strlen(src_cont) + 1;
+			*src_str = strdup(*src_dfs_path);
+		}
+		*src_str_len = strlen(*src_str);
+	} else if (strcmp(src_type, "path") == 0) {
+	        rc = duns_resolve_path(src, src_dattr);
+		if (rc != 0) {
+			/* fs_copy will later check this is a valid src path
+			 * with stat
+			 */
+		  	*src_str = strdup(src);
+		    	src_file_dfs->type = POSIX;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs